server/src/server.js

```diff
@@ -0,0 +1,156 @@
+const { v4: uuidv4 } = require("uuid");
+const express = require("express");
+const cookieSession = require("cookie-session");
+const datastore = require("./datastore");
+const app = express();
+
+const KIND_EVENT = "Event";
+const URL_PARAM_EVENT_ID = `eventID`;
+
+const ERROR_BAD_DB_INTERACTION = "BAD_DATABASE";
+const ERROR_INVALID_EVENT_ID = "INVALID_EVENT_ID";
+const ERROR_BAD_UUID = "BAD_UUID";
+
+// Parse request bodies with the json content header into JSON
+app.use(express.json());
+
+app.use(
+  cookieSession({
+    name: "session",
+    keys: ["secret1"],
+    maxAge: 365 * 24 * 60 * 60 * 1000, // 1 year
+  })
+);
+
+app.use((req, res, next) => {
+  if (req.session.isNew) {
+    // Don't worry about collision
+    const uuid = uuidv4();
+    req.session.userID = uuid;
+  } else {
+    if (
+      !/^[a-z0-9]{8}-[a-z0-9]{4}-4[a-z0-9]{3}-[a-z0-9]{4}-[a-z0-9]{12}$/.test(
```

Comment:
do we care if this is a bad uuid? if so, why?

Reply:
A bad uuid will lead to weird database shenanigans that could go uncaught since users are identified by uuid.

The uuid should never be bad in the normal code flow, but I added this if statement as a way to catch potential bugs in the server code. Specifically, when I was writing the server, my first pass had a bug where the uuid would sometimes be `"undefined"`. Hopefully this if statement will preemptively catch any bugs relating to uuids.

Reply:
ah, makes sense, i'm not used to thinking in terms of JS weak types haha